python/langsmith/client.py

```diff
@@ -4664,6 +4664,553 @@ def _evaluate_strings(
             **kwargs,
         )
 
+    @functools.lru_cache(maxsize=1)
+    def _get_settings(self) -> dict:
```

Comment:
This will impact the expected refcount to this client instance, which would make it so the tracer thread isn't garbage collected.

Would implement this in a similar way to how we handle the `def info()` caching with an explicit private attribute

Reply:
have updated!